app/jobs/reports/base_report.rb

```diff
@@ -54,6 +54,23 @@ def save_report(report_name, body, extension:)
       upload_file_to_s3_timestamped_and_latest(report_name, body, extension)
     end
 
+    def track_report_data_event(event, hash = {})
+      analytics_hash = {
+        name: event,
+        time: Time.zone.now.iso8601,
+      }
+      analytics_hash.merge!(hash)
+      write_hash_to_reports_log(analytics_hash)
```

Comment:
I would remove a few assignments:
```suggestion
      write_hash_to_reports_log({
        name: event,
        time: Time.zone.now.iso8601,
      }.merge(hash))
```